arbor/util/piecewise.hpp

```diff
@@ -382,125 +404,113 @@ namespace impl {
     struct piecewise_pairify {
         std::pair<A, B> operator()(
             double left, double right,
```

Comment:
Arguments `left` and `right` are not used?

Reply:
They're named just for clarity, but this could indeed just be `(double, double, const pw_element<A>& a_elem, const pw_element<B>& b_elem)`.

Reply:
Maybe `[[maybe_unused]]` is helpful to indicate that fact.